packages/components/src/autocomplete/index.js

```diff
@@ -588,10 +369,53 @@ export class Autocomplete extends Component {
 	}
 
 	componentDidUpdate( prevProps, prevState ) {
-		const { open } = this.state;
+		const { record, completers } = this.props;
+		const { record: prevRecord } = prevProps;
 		const { open: prevOpen } = prevState;
-		if ( ( ! open ) !== ( ! prevOpen ) ) {
-			this.toggleKeyEvents( ! ! open );
+
+		if ( ( ! this.state.open ) !== ( ! prevOpen ) ) {
+			this.toggleKeyEvents( ! ! this.state.open );
+		}
+
+		if ( richTextStructure.isCollapsed( record ) ) {
+			const text = richTextStructure.getTextContent( record );
+			const prevText = richTextStructure.getTextContent( prevRecord );
+			const textToSearch = text.slice( 0, record.selection.start );
```

Comment:
I took some time to build and play with this PR and noticed typing "Hello, @-exampl" led to completer's `test` and `getQuery` functions being called with "Hello, @-exampl" instead of "@-exampl". This gives more power to completers, but I wonder whether it would be simpler for authors if we only passed "@-exampl" which is the text between the cursor and the nearest preceding space. What do you think?

Reply:
Yeah, for sure, that sounds good. Not sure about keeping the trigger, let's see when it all works properly. Atm the autocomplete also pops up for something like `text @so|omeone` with an item for e.g. `someone`. We'll need to avoid this somehow. Unsure how this worked before. This PR touches so many components that I'm wondering if you'd be able to help out a bit with the autocomplete component, mainly making sure all current functionality is present?